src/coreclr/vm/debugdebugger.cpp

```diff
@@ -778,6 +777,25 @@ FCIMPL4(void, DebugStackTrace::GetStackFramesInternal,
 }
 FCIMPLEND
 
+extern MethodDesc* QCALLTYPE StackFrame_GetMethodDescFromNativeIP(LPVOID ip)
+{
+    QCALL_CONTRACT;
+
+    MethodDesc* pResult = nullptr;
+
+    BEGIN_QCALL;
+
+    EECodeInfo codeInfo((PCODE)ip);
+    if (codeInfo.IsValid())
+    {
+        pResult = codeInfo.GetMethodDesc();
```

Comment:
This has race condition with unloading that we have discussed over email. The method can get unloaded between the time EECodeInfo constructor looks up the MethodDesc and the time caller converts it to MethodInfo. It should be pretty rare, so I think we can live with this race condition in this temporary method. It may be nice to add TODO about it here so that the problem is easy to root cause if somebody hits it.